Main.py

```diff
@@ -306,13 +306,27 @@ def find_common_pool(players: Set[int], shared_pool: Set[str]) -> Tuple[
     else:
         logger.info("Progression balancing skipped.")
 
-    # we're about to output using multithreading, so we're removing the global random state to prevent accidental use
-    multiworld.random.passthrough = False
-
     if args.skip_output:
         logger.info('Done. Skipped output/spoiler generation. Total Time: %s', time.perf_counter() - start)
         return multiworld
 
+    # determine which items should get starting hints and add those location to start_location_hints to be collected
+    # into multidata later
+    filled_locations = multiworld.get_filled_locations()
+    multiworld.random.shuffle(filled_locations)
```

Comment:
Oh man, that seems kinda slow ngl. We do `filled_locations = multiworld.get_filled_locations()` again later in `write_multidata`.

Would it maybe be possible to get all locations once, then iterate them *in order* and collect **all** instances of a hinted item, and **then** pick a random subset according to the desired count?